### ISSUE TITLE
Add multiple keys support on EXISTS

### ISSUE THREAD

**Issue by kmcheung12** [andymccurdy/redis-py#766]
Redis [EXISTS](http://redis.io/commands/exists) support multiple keys

As of [https://github.com/andymccurdy/redis-py/blob/master/redis/client.py#L939](https://github.com/andymccurdy/redis-py/blob/master/redis/client.py#L939) The python binding currently does not .

Suggest changing 'def exists(self, name):' to 'def exists(self, *name):'

**Comment by thuck** [comment]
Looks like this change would break some code.
The EXISTS is considered a bool in the RESPONSE_CALLBACKS; it should be moved to an int, but this would break anything using ```r.exists(key) is True```.

Also there is  ```__contains__ = exists ```; and this should always return a bool; maybe this can be handled like this:
```
def __contains__(self, *name):
        size = len(names)
        size_from_redis = self.exists(*names)
        if size == size_from_redis:
            return True

        return False
```
@andymccurdy what do you think about the breaking compatibility? Or do you have any idea how to avoid it?

**Comment by kute** [comment]
it could be done by yourself.

    def exists(self, *keys):
        return [redis.exists(key) for key in keys]

or

    def exists(self, *keys):
        with redis.pipeline() as pipe:
             for key in keys:
                 pipe.exists(key)
             return pipe.execute()

**Comment by dgouldin** [comment]
I'm not sure if this will be helpful to anybody else, but since I didn't need backwards compatibility with the `is` operator, I created a client subclass that does what I want:

```
from redis.client import Redis as BaseRedis


class Redis(BaseRedis):
    RESPONSE_CALLBACKS = dict(
        BaseRedis.RESPONSE_CALLBACKS,
        EXISTS=int,
    )

    def exists(self, *names):
        return self.execute_command('EXISTS', *names)
```

```
>>> client = Redis()
>>> client.exists('foo', 'bar', 'baz')
0
>>> client.set('foo', 1)
True
>>> client.exists('foo', 'bar', 'baz')
1
>>> 'foo' in client
True
>>> 'bar' in client
False
```

**Comment by alvinr** [comment]
+1 for this, although EXISTS command returns the *number* of keys that exists, its not a boolean.

```Since Redis 3.0.3 it is possible to specify multiple keys instead of a single one. In such a case, it returns the total number of keys existing. Note that returning 1 or 0 for a single key is just a special case of the variadic usage, so the command is completely backward compatible.```

**Comment by ShaiWavesAudio** [comment]
Current implementation is wrong as EXISTS return an integer not a bool. 
Even calling `execute_command("EXISTS",  "A", "B", "C") ` does not work because RESPONSE_CALLBACKS specify EXISTS as returning bool and convert anything > 0 to True.
So there is no way to get the correct behavior. In my view correct implementation is more important than backward compatibility. Maybe add another method exists_multi() that will handle multiple keys correctly?

**Comment by polsar88** [comment]
If you want `exists_count` to work with a pipeline, do not restore the callback:

```
def exists_count(self, *names):
    return self.execute_command('EXISTS', *names)
redis.StrictRedis.RESPONSE_CALLBACKS['EXISTS'] = int
setattr(redis.StrictRedis, exists_count.__name__, exists_count)
```

**Comment by andymccurdy** [comment]
fixed in redis-py 3.0
